### PR TITLE
#22 simplify / streamline pipelines setup and run process

### PR DESCRIPTION
**Pull request by bob-bn**
This PR is supposed to close/solve ticket #22. Additional details would be mentioned in each of the commit messages separately.

### CODE REVIEW

**Comment by bob-bn** [comment]
@mattigrthr in order to test the changes done ( so far ), you can simply run the following ( will be documented as well ):

- change directory to `kuwala-pipelines`
- run the following command: ` docker-compose --profile pipeline_essentials up -d`
- run the following command:  ` docker-compose run osm-poi`

**Comment by mattigrthr** [comment]
@bob-bn, I found a bug on this branch. The download of new population files fails. I didn't realize it before because I was using an existing download for a country. 

This is the error:
```
[y] IL - Israel
[z] IT - Italy
[0] Next page

Please select a country! [1...9, a...z, 0]: z
Downloading data for Italy
Error: No data available for ITA
    at /opt/population-density/src/utils/countryPicker/index.js:19:17
    at new Promise (<anonymous>)
    at downloadFiles (/opt/population-density/src/utils/countryPicker/index.js:7:12)
    at /opt/population-density/src/utils/countryPicker/index.js:86:23
    at new Promise (<anonymous>)
    at Object.pick (/opt/population-density/src/utils/countryPicker/index.js:41:12)
    at /opt/population-density/src/data/processor/index.js:101:49
    at new Promise (<anonymous>)
    at Object.start (/opt/population-density/src/data/processor/index.js:99:12)
    at /opt/population-density/src/data/main.js:43:41
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

I believe that the shell command cannot be executed in the container. This is where the error is thrown (`kuwala-pipelines/population-density/src/utils/countryPicker/index.js`):
```
console.info(
    `Downloading data for ${countries.getName(selectedCountry, 'en')}`
);

if (
    shell.exec(
        `aws s3 ls --no-sign-request s3://dataforgood-fb-data/csv/month=2019-06/country=${selectedCountry}/`,
        { silent: true }
    ).code !== 0
) {
    return reject(
        new Error(`No data available for ${selectedCountry}`)
    );
}
```

**Comment by bob-bn** [comment]
```
  # The below command builds the images linked to each of the services mentioned
  
  docker-compose build osm-poi population-density 

  # Each of the below commands (Added additional param for the run command) runs the services 
  # in Download or API mode in a blocking way. Add -d to make it a background command
  
  docker-compose run  --service-ports osm-poi start-processing:local
  docker-compose run  --service-ports osm-poi start-api:local
  docker-compose run  --service-ports population-density start-processing:local
  docker-compose run  --service-ports population-density start-api:local

```

**Comment by mattigrthr** [comment]
Is it possible to automatically delete the run containers after they finished? Docker Desktop gets quite crowded otherwise. 😅
image

**Comment by bob-bn** [comment]
> Is it possible to automatically delete the run containers after they finished? Docker Desktop gets quite crowded otherwise. 😅
> image

yes absolutely, we can embed the docker-compose up command in a blocking shell script, which on EXIT would actually trigger a docker-compose down command that will clean up everything.

**Comment by mattigrthr** [comment]
Can you also edit the torrc file? There you have to set how often the IP address is reset. The default is 60s and the lowest you can do is 10s. So we always do the lowest. :stuck_out_tongue: Otherwise requests might fail from time to time.
This is how I do it on an EC2 instance.
```
Set MaxCircuitDirtiness
cd ../../../etc/tor
sudo chmod 777 torrc 
echo 'MaxCircuitDirtiness 10' >> torrc
```

**Comment by bob-bn** [comment]
@mattigrthr All issues are resolved ( including the cleanup for docker and all ) 

- RUN `chmod  +x ./shared/scripts/init.sh`
- RUN `./shared/scripts/init.sh`

This will initialize things up, once you ctrl + c from it ( only once ) it will clean up everything including the intermediate docker-compose run that you will execute.

The issue regarding downloading countries is solved as well ( as discussed it was an awscli missing issue in the docker container )

Added the tor MaxCircuitDirtiness using the ENV TOR_MaxCircuitDirtiness and it is set to 10...however I am not sure if this or something else is causing the google-poi to crash every 3 - 4 search requests. have you encountered this before? Nothing in the logs the app simply shuts down. 

This is the only thing remaining now to figure out why google-poi dockers exist every 3-4 search requests.

**Comment by mattigrthr** [comment]
It works like a charm! 👌🏽 Just for the documentation: Please specify that you need to be in the `kuwala-pipelines` directory to run the script.

File: kuwala-pipelines/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+---
+
+## Prerequisites
+
+Installed version of *Docker* and *docker-compose*
+
+An easy way to run docker on your machine is using [*Docker Desktop*](https://docs.docker.com/desktop/) with a GUI to 
+manage your containers.
+
+#### Hints for macOS
```
Comment (mattigrthr) [inline kuwala-pipelines/README.md:11]: This is obsolete now, right?

**Review by mattigrthr — APPROVED** [review]

**Pull request merged**